qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -97,16 +139,19 @@ def _compute_gradients(
         for exc in self._excitation_pool:
             # add next excitation to ansatz
             self._ansatz.operators = self._excitation_list + [exc]
-            # set the current ansatz
-            vqe.ansatz = self._ansatz
-            # evaluate energies
-            parameter_sets = theta + [-self._delta] + theta + [self._delta]
-            energy_evaluation = vqe.get_energy_evaluation(self._main_operator)
-            energy_results = energy_evaluation(np.asarray(parameter_sets))
+            # the ansatz needs to be decomposed for the gradient to work
+            vqe.ansatz = self._ansatz.decompose()
+            param_sets = sorted(vqe.ansatz.parameters, key=lambda p: p.name)
+            # zip will only iterate the length of the shorter list
+            theta1 = dict(zip(vqe.ansatz.parameters, theta))
+            op = vqe.construct_expectation(theta1, self._main_operator)
             # compute gradient
-            gradient = (energy_results[0] - energy_results[1]) / (2 * self._delta)
-            res.append((np.abs(gradient), exc))
-
+            state_grad = self.gradient.convert(operator=op, params=param_sets)
+            # Assign the parameters and evaluate the gradient
+            value_dict = {param_sets[-1]: 0.0}
+            state_grad_result = state_grad.assign_parameters(value_dict).eval()
```

Comment:
```suggestion
            state_grad_result = sampler.convert(state_grad, params=value_dict).eval()
```
This change is necessary in order to respect the VQE backend. If we do not use a sampler (like we did before) we would always use statevector evaluations.